.github/workflows/push-ci.yml

```diff
@@ -15,58 +15,48 @@ on:
     with:
       checkout-repo: ${{ github.event.pull_request.head.repo.full_name }}
       checkout-ref: ${{ github.event.pull_request.head.ref }}
-    secrets: inherit
-    if: github.repository_owner == 'cryostatio'
-  build-and-test-amd64:
+
+  build-and-test:
     uses: ./.github/workflows/ci-build-image.yml
-    secrets: inherit
-    with:
-      build-arch: amd64
     if: github.repository_owner == 'cryostatio'
-  build-and-test-arm64:
-    uses: ./.github/workflows/ci-build-image.yml
-    secrets: inherit
+    strategy:
+      matrix:
+        arch: [amd64, arm64]
+    permissions:
+      pull-requests: write
     with:
-      build-arch: arm64
-    if: github.repository_owner == 'cryostatio'
+      build-arch: ${{ matrix.arch }}
+
   push-to-quay:
     runs-on: ubuntu-latest
-    needs: [code-analysis, build-and-test-amd64, build-and-test-arm64]
+    needs: [code-analysis, build-and-test]
+    strategy:
+      matrix:
+        arch: [amd64, arm64]
     env:
       CRYOSTAT_IMG: quay.io/cryostat/cryostat
-    if: always() && github.repository_owner == 'cryostatio'
+    permissions:
+      packages: write
     steps:
     - uses: actions/download-artifact@v3
       with:
-        name: cryostat-amd64
-      if: github.repository_owner == 'cryostatio'
-    - name: Load cryostat-amd64 image
-      env:
-        IMAGE_VERSION: ${{ needs.build-and-test-amd64.outputs.image-version }}
-      run: |
-          podman load -i cryostat-amd64.tar
-          podman tag quay.io/cryostat/cryostat:latest $CRYOSTAT_IMG:$IMAGE_VERSION-linux-amd64
-      if: github.repository_owner == 'cryostatio'
-    - uses: actions/download-artifact@v3
-      with:
-        name: cryostat-arm64
-      if: github.repository_owner == 'cryostatio'
-    - name: Load cryostat-arm64 image
+        name: cryostat-${{ matrix.arch }}
+    - name: Load cryostat-${{ matrix.arch }} image
       env:
-        IMAGE_VERSION: ${{ needs.build-and-test-arm64.outputs.image-version }}
+        IMAGE_VERSION: ${{ needs.build-and-test.outputs.image-version }}
       run: |
-          podman load -i cryostat-arm64.tar
-          podman tag quay.io/cryostat/cryostat:latest $CRYOSTAT_IMG:$IMAGE_VERSION-linux-arm64
-      if: github.repository_owner == 'cryostatio'
+          podman load -i cryostat-${{ matrix.arch }}.tar
+          podman tag quay.io/cryostat/cryostat:latest $CRYOSTAT_IMG:$IMAGE_VERSION-linux-${{ matrix.arch }}
     - name: Create OCI Manifest
       id: create-manifest
       env:
-        IMAGE_VERSION: ${{ needs.build-and-test-amd64.outputs.image-version }}
+        IMAGE_VERSION: ${{ needs.build-and-test.outputs.image-version }}
       run: |
-        podman manifest create $CRYOSTAT_IMG:$IMAGE_VERSION containers-storage:$CRYOSTAT_IMG:$IMAGE_VERSION-linux-amd64 containers-storage:$CRYOSTAT_IMG:$IMAGE_VERSION-linux-arm64
+        podman manifest create $CRYOSTAT_IMG:$IMAGE_VERSION containers-storage:$CRYOSTAT_IMG:$IMAGE_VERSION-linux-${{ matrix.arch }}
```

Comment:
I'm not sure this is going to work as expected - I think the matrix setup might make this `podman manifest create` command run two separate times (once for each arch value), but it really needs to run once with all of the arches in a list like it was before.